dozer/cogs/shortcuts.py

```diff
@@ -129,29 +130,52 @@ async def list(self, ctx):
     """
 
     @Cog.listener()
-    async def on_message(self, msg):
+    async def on_message(self, msg: discord.Message):
         """prefix scanner"""
         if not msg.guild or msg.author.bot:
             return
-        setting = await self.settings_cache.query_one(guild_id=msg.guild.id)
-        if setting is None:
-            return
-
-        c = msg.content
-        if len(c) < len(setting.prefix):
-            return
 
-        if not c.startswith(setting.prefix):
+        setting = await ShortcutSetting.get_unique_by(guild_id = msg.guild.id)
+        if not setting:
             return
 
-        shortcuts = await ShortcutEntry.get_by(guild_id=msg.guild.id)
-        if not shortcuts:
-            return
-
-        for shortcut in shortcuts:
-            if c.lower()[len(setting.prefix):] == shortcut.name.lower():
-                await msg.channel.send(shortcut.value)
-                return
+        # Search for the prefix within the message content
+        prefix = setting.prefix
+        prefix_index = msg.content.find(prefix)
+
+        # Check for the presence of the prefix anywhere in the message
+        prefix_index = msg.content.find(prefix)
+
+        if prefix_index != -1:
+            # Ensure there's no space directly after the prefix
+            if prefix_index + len(prefix) < len(msg.content) and msg.content[prefix_index + len(prefix)] == ' ':
+                return  # there's a space, so it was probably meant to be used in text rather than call a shortcut
+
+            # Extract the command part from the message directly after the prefix
+            start_index = prefix_index + len(prefix)
+            remaining_content = msg.content[start_index:].strip()
+
+            # Extract the command name before any spaces or punctuation
+            command_name = remaining_content.split()[0] if remaining_content.split() else ""
+
+            # Check if the extracted command name matches any valid shortcuts
+            all_shortcuts = await ShortcutEntry.get_by(guild_id = msg.guild.id)
+            all_shortcuts = [s.name for s in all_shortcuts]
+
+            if command_name in all_shortcuts:
+                shortcut = await ShortcutEntry.get_unique_by(guild_id = msg.guild.id, name = command_name)
```

Comment:
If we change to use the full shortcut entity in the loop, we do not need to do more database calls. even if we needed to, this should only be done in the case where we actually need the shortcut value.